src/test/kotlin/org/jetBrains/research/tasktracker/tracking/TaskFileHandlerTest.kt

```diff
@@ -0,0 +1,163 @@
+
+    private fun getPrivateFunc(name: String): KFunction<*> {
+        val method = TaskFileHandler::class.memberFunctions.find { it.name == name }
+            ?: error("Function hasn't been found")
+        method.isAccessible = true
+        return method
+    }
+
+    override fun setUp() {
+        super.setUp()
+        taskFileHandler = TaskFileHandler
+    }
+
+    fun testInitDisposeTask() {
+        val projectToTaskToFiles =
+            taskFileHandler.javaClass.getDeclaredField("projectToTaskToFiles").also { it.isAccessible = true }
+                .get(taskFileHandler) as MutableMap<*, MutableMap<*, *>>
+        assert(projectToTaskToFiles.isEmpty())
+
+        assertNoThrowable {
+            taskFileHandler.disposeTask(project, task1)
+            taskFileHandler.initTask(project, task1)
+        }
+        assert(projectToTaskToFiles.size == 1)
+        assert(projectToTaskToFiles[project]?.let { it.size == 1 } ?: false)
+        assertNoThrowable {
+            taskFileHandler.initTask(project, task2)
+        }
+        assert(projectToTaskToFiles.size == 1)
+        assert(projectToTaskToFiles[project]?.let { it.size == 2 } ?: false)
+        assertFailsWith<Throwable> {
+            taskFileHandler.initTask(project, task2)
+        }
+        assert(projectToTaskToFiles[project]?.let { it.size == 2 } ?: false)
+        assertNoThrowable {
+            taskFileHandler.disposeTask(project, task1)
+        }
+        assert(projectToTaskToFiles[project]?.let { it.size == 1 } ?: false)
+        assert(projectToTaskToFiles[project]?.let { it.keys.first() == task2 } ?: false)
+        assertNoThrowable {
+            taskFileHandler.disposeTask(project, task2)
+        }
+        assert(projectToTaskToFiles.isEmpty())
+    }
+
+    fun testAddVirtualFileListener() {
+        val file1 = File("${project.basePath}/test1")
+        val file2 = File("${project.basePath}/test2")
+        val directory = File("${project.basePath}/tests/")
+        ApplicationManager.getApplication().runWriteAction {
+            FileUtil.createIfDoesntExist(file1)
+            FileUtil.createIfDoesntExist(file2)
+            FileUtil.createDirectory(directory)
+        }
+        val virtualFile1 = LocalFileSystem.getInstance().findFileByIoFile(file1) ?: error("file must exist")
+        val virtualFile2 = LocalFileSystem.getInstance().findFileByIoFile(file2) ?: error("file must exist")
+        val virtualDirectory =
+            LocalFileSystem.getInstance().findFileByIoFile(directory) ?: error("file must exist")
+        assert(virtualFile1.exists())
+        assert(virtualFile2.exists())
+        assert(virtualDirectory.exists())
+        val method = getPrivateFunc("addVirtualFileListener")
+        assertNoThrowable {
+            method.call(taskFileHandler, virtualFile1)
+            method.call(taskFileHandler, virtualFile2)
+            method.call(taskFileHandler, virtualDirectory)
+        }
```

Comment:
What actually do you check by this and the next tests? I'd propose to call `initTask` and `disposeTask` in these tests and check if the listener really works, e.g. you can change the file and check if this event was logged by the listener and vice versa. Probably you can check it inside other tests which will be created from `testInitDisposeTask`
What do you think?